### PR TITLE
add simple and kerberos auth type for linkis jdbc

### PR DESCRIPTION
**Pull request by CCweixiao**
### What is the purpose of the change
(add simple and kerberos auth type for jdbc.
It is necessary under the needs of different certification environments.
Related issues: #1085 . )

### Brief change log

- Provide multiple authentication-related parameter types;
- Increase the code logic of kerberos authentication.
- add schedule thread pool to refresh kerberos login status

### Verifying this change
Tested in my local environment

### Guidelines for use

https://mp.weixin.qq.com/s/vcFge4BNiEuW-7OC3P-yaw

### CODE REVIEW

File: linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java [hunk 1]
```diff
@@ -160,18 +205,148 @@ public Connection getConnection(Map<String, String> properties) throws SQLExcept
...
+        validateURL(url);
+        return url.trim();
+    }
+
+    private boolean isUsernameAuthType(Map<String, String> properties) {
+        return "USERNAME".equals(getJdbcAuthType(properties));
+    }
+
+    private boolean isKerberosAuthType(Map<String, String> properties) {
+        return "KERBEROS".equals(getJdbcAuthType(properties));
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java:223]: Can USERNAME/KERBEROS/SIMPLE be defined as a constant

File: linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java [hunk 2]
```diff
@@ -160,18 +205,148 @@ public Connection getConnection(Map<String, String> properties) throws SQLExcept
...
+        return properties.getOrDefault("jdbc.auth.type", "USERNAME").trim().toUpperCase();
+    }
+
+    private void createKerberosSecureConfiguration(Map<String, String> properties) {
+        Configuration conf = new Configuration();
+        conf.set(HADOOP_SECURITY_AUTHENTICATION, KERBEROS.toString());
+        UserGroupInformation.setConfiguration(conf);
+        try {
+            if (!UserGroupInformation.isSecurityEnabled()
+                    || UserGroupInformation.getCurrentUser().getAuthenticationMethod() != KERBEROS
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java:236]: It is better to define Kerberos-related refresh and creation as a separate interface, without modifying the previous code file

File: linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java (hunk 1 above)
Comment (CCweixiao) [reply linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java:223]: > 

这个可以的

File: linkis-commons/linkis-hadoop-common/src/main/scala/org/apache/linkis/hadoop/common/utils/KerberosUtils.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static org.apache.hadoop.fs.CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION;
+import static org.apache.hadoop.security.UserGroupInformation.AuthenticationMethod.KERBEROS;
+
+
+public class KerberosUtils {
```
Comment (peacewong) [inline linkis-commons/linkis-hadoop-common/src/main/scala/org/apache/linkis/hadoop/common/utils/KerberosUtils.java:31]: Should be placed in the java directory

File: linkis-commons/linkis-hadoop-common/src/main/scala/org/apache/linkis/hadoop/common/utils/KerberosUtils.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(KerberosUtils.class);
+
+    private KerberosUtils() {
+    }
+
+    private static Configuration createKerberosSecurityConfiguration() {
+        Configuration conf = new Configuration();
```
Comment (peacewong) [inline linkis-commons/linkis-hadoop-common/src/main/scala/org/apache/linkis/hadoop/common/utils/KerberosUtils.java:38]: Should use HDFSUtils.getConfiguration to get the configuration object

File: linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java [hunk 5]
```diff
@@ -160,18 +204,72 @@ public Connection getConnection(Map<String, String> properties) throws SQLExcept
...
+        if (url.startsWith("\"") && url.endsWith("\"")) {
+            url = url.trim();
+            return url.substring(1, url.length() - 1);
         }
+        return url;
     }
 
     public static void main(String[] args) throws Exception {
 //        Pattern pattern = Pattern.compile("^(jdbc:\\w+://\\S+:[0-9]+)\\s*");
-        String url = "jdbc:mysql://xxx.xxx.xxx.xxx:8504/xx?useUnicode=true&amp;characterEncoding=UTF-8&amp;createDatabaseIfNotExist=true";
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/ConnectionManager.java:174]: Should be placed in the Test class

**Pull request merged**